docs/csharp/programming-guide/generics/snippets/GenericWhereConstraints.cs

```diff
@@ -0,0 +1,282 @@
+                yield return current.Data;
+                current = current.Next;
+            }
+        }
+
+        public T FindFirstOccurrence(string s)
+        {
+            Node current = head;
+            T t = null;
+
+            while (current != null)
+            {
+                //The constraint enables access to the Name property.
+                if (current.Data.Name == s)
+                {
+                    t = current.Data;
+                    break;
+                }
+                else
+                {
+                    current = current.Next;
+                }
+            }
+            return t;
+        }
+    }
+    // </Snippet9>
+
+    public interface IEmployee
+    {
+    }
+
+    // <Snippet10>
+    class EmployeeList<T> where T : Employee, IEmployee, System.IComparable<T>, new()
+    {
+        // ...
+    }
+    // </Snippet10>
+
+    // <Snippet12>
+    class Base { }
+    class Test<T, U>
+        where U : struct
+        where T : Base, new()
+    { }
+    // </Snippet12>
+
+    namespace ListExample
+    {
+        // <Snippet13>
+        public class List<T>
+        {
+            public void Add<U>(List<U> items) where U : T {/*...*/}
+        }
+        // </Snippet13>
+    }
+
+    // <Snippet14>
+    //Type parameter V is used as a type constraint.
+    public class SampleClass<T, U, V> where T : V { }
+    // </Snippet14>
+
+    public static class UnmanagedExtensions
+    {
+        // <Snippet15>
+        unsafe public static byte[] ToByteArray<T>(this T argument) where T : unmanaged
+        {
+            var size = sizeof(T);
+            var result = new Byte[size];
+            Byte* p = (byte*)&argument;
+            for (var i = 0; i < size; i++)
+                result[i] = *p++;
+            return result;
+        }
+        // </Snippet15>
+
+        // <Snippet16>
+        public static TDelegate TypeSafeCombine<TDelegate>(this TDelegate source, TDelegate target)
+            where TDelegate : System.Delegate
+            => Delegate.Combine(source, target) as TDelegate;
+        // </Snippet16>
+
+        // <Snippet18>
+        public static Dictionary<int, string> EnumNamedValues<T>() where T : System.Enum
+        {
+            var result = new Dictionary<int, string>();
+            var values = Enum.GetValues(typeof(T));
+
+            foreach (int item in values)
+                result.Add(item, Enum.GetName(typeof(T), item));
+            return result;
+        }
+        // </Snippet18>
+    }
+    public static class GenericWhereConstraints
+    {
+        public static void Examples()
+        {
+            TestStringEquality();
+            TestUnmanaged();
+            TestDelegateCombination();
+            TestEnumValues();
+        }
+
+        // <Snippet11>
+        public static void OpEqualsTest<T>(T s, T t) where T : class
+        {
+            System.Console.WriteLine(s == t);
+        }
+        private static void TestStringEquality()
```

Comment:
```suggestion

        private static void TestStringEquality()
```

Add a line between methods.